### DISCUSSION TITLE
Why add_redact_annot not redacte the whole text?

### DISCUSSION THREAD

**Discussion by Muhammadraafat1** [pymupdf/PyMuPDF#3543] [Looking for help] [answered]
I use add_redact_annot then apply_redaction then insert_text, why it look like that?
I get the rect from these line:
```
            wlist = page.search_for(word)
            for rect in wlist:

```
![image](https://github.com/pymupdf/PyMuPDF/assets/140750834/8c98ab65-7d0d-4d2b-88b6-3c08fc032184)

I tried to make some shift but it affect in another words, so how deal with this?

**Comment by JorjMcKie** [comment, marked as answer]
An image like yours provides limited information only - and forces me to guess what might actually be happening.

Your image **_looks_** like you are redacting OCRed text.
This is always error-prone:
Text bboxes will regularly be imprecise, and the redaction process will not erase old **_text,_** but punch holes in the image at places where the OCR engine is telling us it has encountered text.
